src/backend/distributed/utils/shardinterval_utils.c

```diff
@@ -68,35 +68,40 @@ CompareShardIntervals(const void *leftElement, const void *rightElement,
 	Datum rightDatum = 0;
 	Datum comparisonDatum = 0;
 	int comparisonResult = 0;
+	bool leftHasNull = (!leftShardInterval->minValueExists ||
+						!leftShardInterval->maxValueExists);
+	bool rightHasNull = (!rightShardInterval->minValueExists ||
+						 !rightShardInterval->maxValueExists);
 
 	Assert(typeCompareFunction != NULL);
 
-	/*
-	 * Left element should be treated as the greater element in case it doesn't
-	 * have min or max values.
-	 */
-	if (!leftShardInterval->minValueExists || !leftShardInterval->maxValueExists)
+	if (leftHasNull && rightHasNull)
+	{
+		comparisonResult = 0;
+	}
+	else if (leftHasNull)
 	{
 		comparisonResult = 1;
-		return comparisonResult;
 	}
-
-	/*
-	 * Right element should be treated as the greater element in case it doesn't
-	 * have min or max values.
-	 */
-	if (!rightShardInterval->minValueExists || !rightShardInterval->maxValueExists)
+	else if (rightHasNull)
 	{
 		comparisonResult = -1;
-		return comparisonResult;
 	}
+	else
+	{
+		/* if both shard interval have min/max values, calculate the comparison result */
+		leftDatum = leftShardInterval->minValue;
```

Comment:
super minor: I think we can declare `leftDatum ` and `rightDatum ` inside the else block.